pyrevitlib/pyrevit/revit/__init__.py

```diff
@@ -93,5 +93,72 @@ def get_project_info():
         return query.get_project_info()
 
 
+class DocumentCorruptionException(Exception):
+    pass
+
+
+class FailureSwallowerContext():
+    """Suppresses warnings during script execution
+
+    >>> with FailureSwallowerContext(force=True) as swallower:
+    >>>     for fam in families:
+    >>>         revit.doc.EditFamily(fam)
+    >>>         if swallower.get_swallowed():
+    >>>             logger.warn("Warnings swallowed")
+    """
+    # static, because the event cannot access non-static class variables
+    failures_swallowed = []
```

Comment:
Here is the most unstable part.

I couldn't find a way to store suppressed warnings as class properties and to reach them from `FailuresProcessing ` Event. 

For the same reason all the methods which should be triggered from the event are static.

Maybe you can propose a better solution